### PR TITLE
add subjects to templates

### PR DESCRIPTION
**Pull request by elliothursh**
## Describe your changes
Adds the ability to define email subjects along side an email template, and uses that when sending emails if no subject is passed in.

## Issue link
Part of #348

## Checklist before requesting a review

- [X] I have performed a self-review of my code
- [X] If it is a core feature, I have added thorough tests.

### CODE REVIEW

File: packages/cli/src/components/PreviewSender.tsx [hunk 1]
```diff
@@ -1,13 +1,11 @@
 import React, { useCallback, useEffect, useState } from "react";
 
 type PreviewSenderProps = {
-  html?: string;
-  previewFunction?: string;
-  previewClass?: string;
```
Comment (psugihara) [inline packages/cli/src/components/PreviewSender.tsx:6]: 👍

File: packages/core/src/index.ts [hunk 2]
```diff
@@ -112,9 +112,19 @@ export function buildSendMail<T>(options: BuildSendMailOptions<T>) {
       derivedTemplateName = component.type.name;
       mailOptions.html = renderedHtml;
     }
-
     if (!mailOptions.html) throw new Error("sendMail couldn't find your html");
 
+    // Get subject from the component if not provided
+    if (component && !mailOptions.subject) {
+      if (typeof component.type.subject === "string") {
```
Comment (psugihara) [inline packages/core/src/index.ts:119]: looking good. maybe add a test that hits this case?

**Comment by psugihara** [comment]
let's add this functionality here #366

**Review by psugihara — APPROVED** [review]
looks grest

**Review by alexfarrill — APPROVED** [review]
Nice, it works!

**Pull request merged**